infra/src/main/kotlin/fi/fta/geoviite/api/frameconverter/v1/FrameConverterServiceV1.kt

```diff
@@ -61,115 +59,84 @@ constructor(
+    private fun getTrackNumberInfo(
         trackNumberIds: List<IntId<LayoutTrackNumber>>,
-        layoutContext: LayoutContext,
-    ): Map<IntId<LayoutTrackNumber>, Oid<LayoutTrackNumber>> {
-        return trackNumberDao.fetchExternalIds(layoutContext.branch, trackNumberIds).mapValues { (_, externalId) ->
-            externalId.oid
+        branch: LayoutBranch,
+    ): Map<IntId<LayoutTrackNumber>, TrackNumberDetails> {
+        val trackNumbers = trackNumberService.getMany(branch.official, trackNumberIds)
+        val oids = trackNumberDao.fetchExternalIds(branch, trackNumberIds)
+        return trackNumbers.associate { trackNumber ->
+            val tnId = trackNumber.id as IntId
+            tnId to
+                TrackNumberDetails(trackNumber, oids[tnId]?.oid) {
+                    geocodingService.getGeocodingContext(branch.official, tnId)
+                }
         }
     }
 
     private fun getLocationTrackOids(
+        branch: LayoutBranch,
         locationTrackIds: List<IntId<LocationTrack>>,
-        layoutContext: LayoutContext,
-    ): Map<IntId<LocationTrack>, Oid<LocationTrack>> {
-        return locationTrackDao.fetchExternalIds(layoutContext.branch, locationTrackIds).mapValues { (_, externalId) ->
-            externalId.oid
-        }
-    }
+    ): Map<IntId<LocationTrack>, Oid<LocationTrack>> =
+        locationTrackDao.fetchExternalIds(branch, locationTrackIds).mapValues { (_, externalId) -> externalId.oid }
 
     private fun filterByRequest(
         request: ValidCoordinateToTrackAddressRequestV1,
         track: LocationTrack,
-        locationTrackOids: Map<IntId<LocationTrack>, Oid<LocationTrack>>,
-        trackNumbers: Map<DomainId<LayoutTrackNumber>, LayoutTrackNumber>,
-        trackNumberOids: Map<IntId<LayoutTrackNumber>, Oid<LayoutTrackNumber>>,
+        locationTrackOid: Oid<LocationTrack>?,
+        trackNumberDetails: TrackNumberDetails,
     ): Boolean =
         all(
             // Spatial cache only returns non-deleted tracks -> no need to check the state here
-            {
-                request.locationTrackOid?.let { locationTrackOid -> locationTrackOid == locationTrackOids[track.id] }
-                    ?: true
-            },
-            { request.locationTrackName?.let { locationTrackName -> locationTrackName == track.name } ?: true },
-            { request.locationTrackType?.let { locationTrackType -> locationTrackType == track.type } ?: true },
-            {
-                request.trackNumberOid?.let { trackNumberOid -> trackNumberOid == trackNumberOids[track.trackNumberId] }
-                    ?: true
-            },
-            {
-                request.trackNumberName?.let { trackNumberName ->
-                    val trackNumber = trackNumbers[track.trackNumberId]
-                    trackNumber?.number == trackNumberName
-                } ?: true
-            },
+            { request.locationTrackOid?.let { it == locationTrackOid } ?: true },
```

Comment:
En yleensä suosi it-nimiä mutta näin suoraviivaisissa lambdoissa se on mukavan lyhyt ja johtaa että tämä kokonaisuus pysyy selkeän luettavina yksi-rivisinä steppeinä.